ui/src/localPeer.ts

```diff
@@ -0,0 +1,61 @@
+import * as remote from "./remote";
+
+// TYPES
+export enum StatusType {
+  Stopped = "stopped",
+  Offline = "offline",
+  Started = "started",
+  Syncing = "syncing",
+  Online = "online",
+}
+
+interface Stopped {
+  kind: StatusType.Stopped;
+}
+
+interface Offline {
+  kind: StatusType.Offline;
+}
+
+interface Started {
+  kind: StatusType.Started;
+}
+
+interface Syncing {
+  kind: StatusType.Syncing;
+  syncs: number;
+}
+
+interface Online {
+  kind: StatusType.Syncing;
+  connected: number;
+}
+
+type Status = Stopped | Offline | Started | Syncing | Online;
+
+enum EventKind {
+  StatusChanged = "LOCAL_PEER_STATUS_CHANGED",
+}
+
+interface StatusChanged {
+  kind: EventKind.StatusChanged;
+  status: Status;
+}
+
+export type PeerEvent = StatusChanged;
+
+// STATE
+const statusStore = remote.createStore<Status>();
+export const status = statusStore.readable;
+
+statusStore.start(() => {
```

Comment:
Should we call this in `App.svelte`?

Reply:
I think it's reasonable to only set up the subscription when at least one subscriber exists for the store. This way it doesn't require a global set up, which can be forgotten.

Reply:
It makes it more self-contained.

Reply:
Ah, I think I misunderstood what this was doing initially. Never mind.